src/lib.rs

```diff
@@ -1,3 +1,29 @@
+//! # Google Encoded Polyline encoding & decoding in Rust
+//!
+//! [Polyline](https://developers.google.com/maps/documentation/utilities/polylinealgorithm)
+//! is a lossy compression algorithm that allows you to store a series of coordinates as a
+//! single string.
+//!
+//! The encoding process converts a binary value into a series of character codes for ASCII
+//! characters using the familiar base64 encoding scheme: to ensure proper display of these
+//! characters, encoded values are summed with 63 (the ASCII character '?') before converting
+//! them into ASCII. The algorithm also checks for additional character codes for a given
+//! point by checking the least significant bit of each byte group; if this bit is set to 1,
+//! the point is not yet fully formed and additional data must follow.
```

Comment:
In my opinion, considering you already have a hyperlink to the Polyline algorithm above that details how it works, I don't think we need this paragraph here. It's not technical enough to refer to it when implementing the algorithm, and it's not high level enough that it warrants being the second paragraph the user reads. Thoughts?

Reply:
Hey, sure now when rereading the second paragraph it seems redundant since I already included the hyperlink.
I will omit the paragraph, also do You have any further suggestions regarding the crate documentation or is the first paragraph and the example sufficient?

Reply:
I think the first paragraph and the example are sufficient – thanks again for doing this!